pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/RuntimeSpawner.java

```diff
@@ -82,7 +83,8 @@ public void start() throws Exception {
             processLivenessCheckTimer.scheduleAtFixedRate(new TimerTask() {
                 @Override
                 public void run() {
-                    if (!runtime.isAlive()) {
+                    Runtime runtime = RuntimeSpawner.this.runtime;
```

Comment:
When does runtime become null?

Reply:
if you close the runtime first and cancel the timer later. that's why I also changed the order of closing sequence.